### DISCUSSION TITLE
`Dependency review` is unexpected and undesired

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#106884] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

I'm looking at: https://github.com/octokit/request.js/pull/669/commits/f3a70079c6a3401d9ba5e8e24f7392b145de6a23#diff-f84e810a117ccd70ff03458a46f6563fb4c2d46f8ec8ef633a90c55a2466aaeb

![image](https://github.com/community/community/assets/2119212/6f84d92e-bd79-4989-a0f0-be23f93e14a5)

There's a button which _normally_ would show a pretty view of a thing, but when I clicked it for a yaml file, I got:

https://github.com/octokit/request.js/pull/669/commits/f3a70079c6a3401d9ba5e8e24f7392b145de6a23?short_path=f84e810#diff-f84e810a117ccd70ff03458a46f6563fb4c2d46f8ec8ef633a90c55a2466aaeb

![image](https://github.com/community/community/assets/2119212/c0884b84-455c-4182-84b2-8c3195c52628)

No dependencies changed.
The changes to this file likely do not affect the dependencies

---

I'm not sure what it's thinking and I'm too exhausted to try, but it didn't make me happy.

**Comment by juxtin** [comment]
Hi @jsoref, when you click the rich diff button on an Actions workflow file, it tries to display any changes in the Actions that are used by that workflow. In this case, since what changed was a parameter and not an Action itself, there's nothing for it to display. I hope that answers your question!

**Reply by jsoref** [reply]
That does not.

A button labeled "rich diff" is not a good way to say "dependency diff".

If the button is only going to show a dependency diff, it should say so. I am not a diviner. Most users are not diviners.

**Reply by jonjanego** [reply]
Hi @jsoref if you are changing a markdown file, clicking the rich diff button will render the markdown, making it easier to compare changes. Since you're examining a yml file there is nothing to render in that markdown viewer.

**Reply by jsoref** [reply]
Yes, I'm clearly used to clicking that button for markdown files and getting a helpful diff.

When I am reviewing files that _don't_ have a pretty renderer, I don't get a button. And that doesn't bother me at all.

1. I'd much prefer to have buttons only display when they have something to do.
2. I'd much prefer it tell me what it will do (here "dependency diff") -- especially if had I changed both some values and also changed a dependency at the same time -- it would be very surprising for me for a "rich" diff to show a _subset_ of all my changes.

Consider:
https://github.com/github/codeql/pull/15493/files#diff-21e0d8ff8b280b1cce8085af4934d04610be13cde25165f01c4623c9ad241090

![image](https://github.com/community/community/assets/2119212/4d58c502-56ff-4cc4-a1b9-f11286e1151f)

This is _not_ a **rich** diff:
![image](https://github.com/community/community/assets/2119212/73b5bfd8-5ffd-4473-8012-313a83b9099c)

It's a _poor_ diff. It lost a really important piece of information -- namely the permissions being changed by the PR. (In some instances I'm adding, and in some instances I'm removing.)

**Comment by ettaboyle** [comment]
@jsoref 👋🏻 Just a heads up that based on feedback from a colleague I've moved this post from the `Code Security` category to `Pull Requests`, and updated this as `Product Feedback` in the labels instead of `Bug`. No further action is needed right now, just wanted to let you know!
